packages/integration-tests/src/amplify_app_pool.ts

```diff
@@ -46,114 +47,124 @@ class DefaultAmplifyAppPool implements AmplifyAppPool {
   }
 
   fetchTestBranchDetails = async (testBranch: TestBranch): Promise<Branch> => {
-    const branch = (
-      await this.amplifyClient.send(
-        new GetBranchCommand({
-          appId: testBranch.appId,
-          branchName: testBranch.branchName,
-        })
-      )
-    ).branch;
-    if (!branch) {
-      throw new Error(
-        `Failed to retrieve ${testBranch.branchName} branch of app ${testBranch.appId}`
-      );
-    }
-    return branch;
+    return this.retryableOperation(async () => {
+      const branch = (
+        await this.amplifyClient.send(
+          new GetBranchCommand({
+            appId: testBranch.appId,
+            branchName: testBranch.branchName,
+          })
+        )
+      ).branch;
+      if (!branch) {
+        throw new Error(
+          `Failed to retrieve ${testBranch.branchName} branch of app ${testBranch.appId}`
+        );
+      }
+      return branch;
+    });
   };
 
   createTestBranch = async (): Promise<TestBranch> => {
-    const app = await this.getAppWithCapacity();
-    const branch = (
-      await this.amplifyClient.send(
-        new CreateBranchCommand({
-          branchName: `${this.testBranchPrefix}${shortUuid()}`,
+    return this.retryableOperation(async () => {
+      const app = await this.getAppWithCapacity();
+      const branch = (
+        await this.amplifyClient.send(
+          new CreateBranchCommand({
+            branchName: `${this.testBranchPrefix}${shortUuid()}`,
+            appId: app.appId,
+          })
+        )
+      ).branch;
+      if (app.appId && branch?.branchName) {
+        const testBranch: TestBranch = {
           appId: app.appId,
-        })
-      )
-    ).branch;
-    if (app.appId && branch?.branchName) {
-      const testBranch: TestBranch = {
-        appId: app.appId,
-        branchName: branch.branchName,
-      };
-      this.branchesCreated.push(testBranch);
-      return testBranch;
-    }
+          branchName: branch.branchName,
+        };
+        this.branchesCreated.push(testBranch);
+        return testBranch;
+      }
 
-    throw new Error('Unable to create branch');
+      throw new Error('Unable to create branch');
+    });
   };
 
   private listAllTestAmplifyApps = async (): Promise<Array<App>> => {
-    let nextToken: string | undefined = undefined;
-    const apps: Array<App> = [];
-    do {
-      const listAppsCommandOutput: ListAppsCommandOutput =
-        await this.amplifyClient.send(
-          new ListAppsCommand({
-            maxResults: 100,
-            nextToken,
-          })
-        );
-      nextToken = listAppsCommandOutput.nextToken;
-      listAppsCommandOutput.apps
-        ?.filter((app: App) => app.name?.startsWith(this.testAppPrefix))
-        .forEach((app: App) => {
-          apps.push(app);
-        });
-    } while (nextToken);
-    return apps;
+    return this.retryableOperation(async () => {
```

Comment:
it is enough to wrap top level public methods in retry.
We shouldn't be applying this aggressively. 

please remove this from private methods for now.